src/redturtle/prenotazioni/browser/bookings_export.py

```diff
@@ -49,25 +53,62 @@ def csv_fields(self):
     def csv_filename(self):
         """Return a filename for this csv"""
 
-        return "%s-%s.csv" % (self.filename, self.date.date().isoformat())
+        return "%s_%s.csv" % (
+            self.filename,
+            self.booking_start_from.date().isoformat()
+            + "-"
+            + self.booking_start_to.date().isoformat(),
+        )
 
     @property
     def brains(self):
+        created = (self.booking_creation_from or self.booking_creation_to) and {
+            "query": (self.booking_creation_from and self.booking_creation_to)
+            and (
+                get_default_timezone(True).localize(self.booking_creation_from),
+                get_default_timezone(True).localize(self.booking_creation_to),
+            )
+            or self.booking_creation_from
+            and get_default_timezone(True).localize(self.booking_creation_from)
+            or self.booking_creation_to
+            and get_default_timezone(True).localize(self.booking_creation_to),
+            "range": (self.booking_creation_from and self.booking_creation_to)
+            and "min:max"
+            or self.booking_creation_from
+            and "min"
+            or self.booking_creation_to
+            and "max",
+        }
 
-        return api.portal.get_tool("portal_catalog").unrestrictedSearchResults(
-            portal_type="Prenotazione",
-            Date={
-                "query": (
-                    get_default_timezone(True).localize(self.date),
-                    get_default_timezone(True).localize(
-                        self.date.replace(hour=23, minute=59)
+        if created:
+            return api.portal.get_tool("portal_catalog").unrestrictedSearchResults(
+                portal_type="Prenotazione",
+                Date={
```

Comment:
Why search on Date index? 
Date is effective date or modified date (if item is not published yet).
Shouldn't be better searching for booking date? Or Date index is filled with booking date for bookings?

Reply:
is filled with booking date